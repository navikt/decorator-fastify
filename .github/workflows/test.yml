name: test

on:
  pull_request:
    branches: 
      - main

env:
  IMAGE: ghcr.io/${{ github.repository }}:${{ github.sha }}
jobs:
  test:
    name: test
    runs-on: ubuntu-latest
    permissions:
      contents: read
      id-token: write
      packages: write
    steps:
      - uses: actions/checkout@v3
      - uses: oven-sh/setup-bun@v1
        with:
          bun-version: '1.0.1'
      - name: Install dependencies
        run: bun install --immutable
        env:
          NODE_AUTH_TOKEN: ${{ secrets.READER_TOKEN }}
      - name: Build app
        run: bun run build
      - name: Run tests
        run: bun test