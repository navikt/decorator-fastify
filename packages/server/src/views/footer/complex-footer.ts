import { LinkGroup, Texts } from 'decorator-shared/types';
import cls from 'decorator-shared/utilities.module.css';
import html from 'decorator-shared/html';
import { ScreenshareButton } from './screenshare-button';
import { ArrowUp } from 'decorator-shared/views/icons';

import classes from 'decorator-client/src/styles/complex-footer.module.css';
import { FooterLenke } from './lenke';
import { Features } from '../../unleash-service';

export type ComplexFooterProps = {
  texts: Pick<Texts, 'share_screen' | 'to_top'>;
  links: LinkGroup[];
  features: Features;
};

export function ComplexFooter({ texts, links, features }: ComplexFooterProps) {
  const isScreensharingEnabled = features['dekoratoren.skjermdeling'];

  return html`
    <footer class="${classes.footer}" data-theme="dark">
      <div class="${classes.footerContent} ${cls.contentContainer}">
        <a class="${classes.toTopLink}" href="#">
          ${ArrowUp()} ${texts.to_top}
        </a>

        <div class="${classes.footerLinks}">
          <ul class="${classes.footerLinkList}">
            ${links.map(
              ({ heading, children }) => html`
                <li class="${classes.footerLinkGroup}">
                  ${heading &&
                  html`<h2 class="${classes.footerLinkHeading}">
                    ${heading}
                  </h2>`}
                  <ul class="${classes.footerInnerLinkList}">
                    ${children.map(
                      (link) => html`
                        <li>
                          ${FooterLenke({
                            link,
                            classNameOverride: classes.footerLink,
                          })}
                        </li>
                      `,
                    )}
                  </ul>
                </li>
              `,
            )}
            ${isScreensharingEnabled &&
            html`<li>${ScreenshareButton(texts.share_screen)}</li>`}
          </ul>
        </div>

        <div class="${classes.complexFooterOrg}">
          <img src="/public/ikoner/meny/nav-logo-white.svg" alt="NAV-logo" />
          <span>Arbeids- og velferdsetaten</span>
        </div>
      </div>
    </footer>
  `;
}
