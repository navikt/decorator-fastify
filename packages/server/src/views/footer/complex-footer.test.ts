import { expect, test } from 'bun:test';
import { ComplexFooter } from './complex-footer';
import UnleashService from '../../unleash-service';

const unleashService = new UnleashService({ mock: true, env: 'development' });

const links = [
  {
    heading: 'Kontakt',
    children: [
      {
        content: 'Kontakt oss',
        url: '/person/kontakt-oss/nb',
      },
      {
        content: 'NAV i ditt fylke',
        url: '/no/nav-og-samfunn/kontakt-nav/nav-i-ditt-fylke',
      },
      {
        content: 'Kurs fra NAV',
        url: '/no/nav-og-samfunn/kontakt-nav/kurs-fra-nav',
      },
      {
        content: 'Klage, tilbakemelding, ros',
        url: '/person/kontakt-oss/tilbakemeldinger',
      },
    ],
  },
  {
    heading: 'Nyheter og presse',
    children: [
      {
        content: 'Nyheter, pressemeldinger og pressekontakt',
        url: '/samarbeidspartner/presse',
      },
    ],
  },
  {
    heading: 'NAV og samfunn',
    children: [
      {
        content: 'Statistikk, analyse og FoU',
        url: '/no/nav-og-samfunn/statistikk-analyse-og-fou',
      },
      {
        content: 'Lover og regler (lovdata.no)',
        url: 'https://lovdata.no/nav/',
      },
      {
        content: 'Om NAV',
        url: '/no/nav-og-samfunn/om-nav/om-nav',
      },
    ],
  },
  {
    heading: 'Andre språk',
    children: [
      {
        content: 'English',
        url: '/en',
      },
      {
        content: 'Sámegiella',
        url: '/se',
      },
    ],
  },
  {
    children: [
      {
        content: 'Personvern og informasjonskapsler',
        url: '/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten',
      },
      {
        content: 'Tilgjengelighet',
        url: '/tilgjengelighet',
      },
    ],
  },
];

test('renders complex footer', async () => {
  expect(
    ComplexFooter({
      texts: {
        to_top: 'Til toppen',
        share_screen: 'Del skjerm med veileder',
      },
      links,
      features: unleashService.getFeatures(),
    }).render(),
  ).toMatchSnapshot();
});
