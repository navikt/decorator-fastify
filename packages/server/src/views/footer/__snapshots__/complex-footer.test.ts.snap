// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`renders complex footer 1`] = `
"
    <footer class="_footer_1vx5s_1" data-theme="dark">
      <div class="_footerContent_1vx5s_38 _contentContainer_1av57_1">
        <a class="_toTopLink_1vx5s_44" href="#">
          
    <svg
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      role="img"
    >
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M12.53 3.47a.75.75 0 0 0-1.06 0l-6.5 6.5a.75.75 0 1 0 1.06 1.06l5.22-5.22V20.5a.75.75 0 0 0 1.5 0V5.81l5.22 5.22a.75.75 0 1 0 1.06-1.06l-6.5-6.5Z"
        fill="currentColor"
      ></path>
    </svg>
   Til toppen
        </a>

        <div class="_footerLinks_1vx5s_53">
          <ul class="_footerLinkList_1vx5s_1">
            
                <li class="_footerLinkGroup_1vx5s_71">
                  <h2 class="_footerLinkHeading_1vx5s_75">
                    Kontakt
                  </h2>
                  <ul class="_footerInnerLinkList_1vx5s_66">
                    
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/person/kontakt-oss/nb"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//person/kontakt-oss/nb&quot;,&quot;label&quot;:&quot;/person/kontakt-oss/nb&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Kontakt oss
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/no/nav-og-samfunn/kontakt-nav/nav-i-ditt-fylke"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//no/nav-og-samfunn/kontakt-nav/nav-i-ditt-fylke&quot;,&quot;label&quot;:&quot;/no/nav-og-samfunn/kontakt-nav/nav-i-ditt-fylke&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    NAV i ditt fylke
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/no/nav-og-samfunn/kontakt-nav/kurs-fra-nav"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//no/nav-og-samfunn/kontakt-nav/kurs-fra-nav&quot;,&quot;label&quot;:&quot;/no/nav-og-samfunn/kontakt-nav/kurs-fra-nav&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Kurs fra NAV
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/person/kontakt-oss/tilbakemeldinger"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//person/kontakt-oss/tilbakemeldinger&quot;,&quot;label&quot;:&quot;/person/kontakt-oss/tilbakemeldinger&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Klage, tilbakemelding, ros
  </a>

                        </li>
                      
                  </ul>
                </li>
              
                <li class="_footerLinkGroup_1vx5s_71">
                  <h2 class="_footerLinkHeading_1vx5s_75">
                    Nyheter og presse
                  </h2>
                  <ul class="_footerInnerLinkList_1vx5s_66">
                    
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/samarbeidspartner/presse"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//samarbeidspartner/presse&quot;,&quot;label&quot;:&quot;/samarbeidspartner/presse&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Nyheter, pressemeldinger og pressekontakt
  </a>

                        </li>
                      
                  </ul>
                </li>
              
                <li class="_footerLinkGroup_1vx5s_71">
                  <h2 class="_footerLinkHeading_1vx5s_75">
                    NAV og samfunn
                  </h2>
                  <ul class="_footerInnerLinkList_1vx5s_66">
                    
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/no/nav-og-samfunn/statistikk-analyse-og-fou"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//no/nav-og-samfunn/statistikk-analyse-og-fou&quot;,&quot;label&quot;:&quot;/no/nav-og-samfunn/statistikk-analyse-og-fou&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Statistikk, analyse og FoU
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="https://lovdata.no/nav/"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt/https://lovdata.no/nav/&quot;,&quot;label&quot;:&quot;https://lovdata.no/nav/&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Lover og regler (lovdata.no)
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/no/nav-og-samfunn/om-nav/om-nav"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//no/nav-og-samfunn/om-nav/om-nav&quot;,&quot;label&quot;:&quot;/no/nav-og-samfunn/om-nav/om-nav&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Om NAV
  </a>

                        </li>
                      
                  </ul>
                </li>
              
                <li class="_footerLinkGroup_1vx5s_71">
                  <h2 class="_footerLinkHeading_1vx5s_75">
                    Andre språk
                  </h2>
                  <ul class="_footerInnerLinkList_1vx5s_66">
                    
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/en"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//en&quot;,&quot;label&quot;:&quot;/en&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    English
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/se"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//se&quot;,&quot;label&quot;:&quot;/se&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Sámegiella
  </a>

                        </li>
                      
                  </ul>
                </li>
              
                <li class="_footerLinkGroup_1vx5s_71">
                  
                  <ul class="_footerInnerLinkList_1vx5s_66">
                    
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten&quot;,&quot;label&quot;:&quot;/no/nav-og-samfunn/om-nav/personvern-i-arbeids-og-velferdsetaten&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Personvern og informasjonskapsler
  </a>

                        </li>
                      
                        <li>
                          
  <a
    is="lenke-med-sporing"
    href="/tilgjengelighet"
    data-analytics-event-args="{&quot;category&quot;:&quot;dekorator-footer&quot;,&quot;action&quot;:&quot;kontakt//tilgjengelighet&quot;,&quot;label&quot;:&quot;/tilgjengelighet&quot;}"
    
    class="_footerLink_1vx5s_1"
    
    
  >
    Tilgjengelighet
  </a>

                        </li>
                      
                  </ul>
                </li>
              
            <li><button type="button" class="_screenshareButton_tarl4_1">
    Del skjerm med veileder
    <svg
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      role="img"
    >
      <path
        fill-rule="evenodd"
        clip-rule="evenodd"
        d="M2.25 4.5c0-.69.56-1.25 1.25-1.25h17c.69 0 1.25.56 1.25 1.25v11c0 .69-.56 1.25-1.25 1.25h-7.75v2.5H19a.75.75 0 0 1 0 1.5H6a.75.75 0 0 1 0-1.5h5.25v-2.5H3.5c-.69 0-1.25-.56-1.25-1.25v-11Zm1.5.25v10.5h16.5V4.75H3.75Z"
        fill="currentColor"
      ></path>
    </svg>
  </button></li>
          </ul>
        </div>

        <div class="_complexFooterOrg_1vx5s_27">
          <img src="/public/ikoner/meny/nav-logo-white.svg" alt="NAV-logo" />
          <span>Arbeids- og velferdsetaten</span>
        </div>
      </div>
    </footer>
  "
`;
